### PR TITLE
[stm32] also compile C++ files

### PR DESCRIPTION
**Pull request by flixr**
only tested with some dummy C++ files, but it should work fine for all.

### CODE REVIEW

**Comment by podhrmic** [comment]
@flixr why do we want to be able to compile C++ files for STM32 autopilots?

**Comment by flixr** [comment]
Why not?
Somebody might want to use a C++ lib (e.g. Eigen) or port something to C++, or ...

**Comment by podhrmic** [comment]
OK, I guess it is more like "useful in the future" feature then. It would be interesting what including C++ files does with code size, but that is a different story...

**Comment by flixr** [comment]
That would strongly depend what C++ features you use...

File: conf/Makefile.stm32 [hunk 1]
```diff
@@ -199,8 +222,8 @@ sym: $(OBJDIR)/$(TARGET).sym
 
 # Link: create ELF output file from object files.
 .SECONDARY : $(OBJDIR)/$(TARGET).elf
-.PRECIOUS : $(COBJ) $(AOBJ)
-%.elf: $(COBJ) $(AOBJ) | $(OBJDIR)
+.PRECIOUS : $(OBJS) $(AOBJ)
+%.elf: $(OBJS) $(AOBJ) | $(OBJDIR)
 	@echo LD $@
 	$(Q)$(LD) $(LDFLAGS) $($(TARGET).LDFLAGS) -o $@ $(COBJ) $(AOBJ) $(LDLIBS)
```
Comment (dewagter) [inline conf/Makefile.stm32]: Whenever a cpp file is compiled, you need to link with g++ instead of gcc

Comment (flixr) [inline conf/Makefile.stm32]: Any reason not to always use g++ for linking?

**Comment by podhrmic** [comment]
@flixr should we merge this? Seems to be minor enough.

**Comment by flixr** [comment]
With this g++ is always used for linking...
I thought that the only difference linking with g++ is that -lstdc++ added and if nothing from libstdc++ is used, it *should* make no difference.

That is *nearly* the case, the lib order also changed:
* gcc: `-lc -lm -lgcc `
* g++: `-lgcc -lstdc++ -lm -lc`

Linking e.g. Quad_LisaMX with g++ actually results in a slightly smaller elf than linking with gcc, probably because the link order changed there are some symbols taken from libcc instead of libc.

**Comment by podhrmic** [comment]
@dewagter @gautierhattenberger should I merge this as is?

**Comment by gautierhattenberger** [comment]
On the principle I'm fine, but I would prefer to test it before merging (okay, could have done that earlier... )

**Comment by podhrmic** [comment]
Merging since it doesn't seem to break anything and having C++ support is useful.

**Pull request merged**